third_party/move/move-compiler-v2/src/lint_common.rs

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+//! This module contains common code useful for lint checkers at various stages
+//! of the compilation pipeline.
+
+use move_compiler::shared::known_attributes::LintAttribute;
+use move_model::{ast::Attribute, model::GlobalEnv};
+use std::str::FromStr;
+use strum_macros::{Display, EnumString};
+
+/// Enumeration of all the lint checks that can be performed.
+///
+/// With the use of `strum_macros::EnumString` and `strum_macros::Display`,
+/// each of the variants can be serialized to and deserialized from a string.
+/// The serialization follows "snake_case".
+#[derive(Copy, Clone, Ord, Eq, PartialEq, PartialOrd, EnumString, Display)]
+#[strum(serialize_all = "snake_case")]
+pub enum LintChecker {
+    BlocksInConditions,
+    UnnecessaryBooleanIdentityComparison,
+    UnnecessaryNumericalExtremeComparison,
+    WhileTrue,
+}
+
+/// Extract all the lint checks to skip from the given attributes.
+/// Also performs error-checking on any `LintAttribute::SKIP` attributes.
+pub fn lint_skips_from_attributes(env: &GlobalEnv, attrs: &[Attribute]) -> Vec<LintChecker> {
```

Comment:
Probably just stay with all what you have here. I was hoping for a more generic treatment in the env, but never mind, you have already build a lot of logic here.

Reply:
We can certainly revisit this - I briefly saw the pragma stuff, but could relate to the unit testing attribute more directly in terms of the features I needed.

I have added several tests to do various checks on the attributes - which would be useful if we switch to placing all this logic elsewhere. It is also similar to what the unit testing plan builder does, so if we do a migration to a new architecture, they can all be done together.

In this mechanism, we potentially skip running a linter if not needed (rather than filtering after the fact), which may be good for more expensive lints.